### ISSUE TITLE
Typescript mixin classes with mobx decorators, how to merge into 1 class?

### ISSUE THREAD

**Issue by RobinRadic** [mobxjs/mobx#1225]
I would like to use 2 or more classes that contain MobX decorators as mixin, like:

```typescript

class DogMeta {
    @observable meta = {
        name: 'foo',
        age : 15
    }
    @action setName(name) {
        this.meta.name = name;
    }
    @action setAge(age) {
        this.meta.age = age;
    }
    @computed 
    get name(): string {return this.meta.name}
    set name(name: string) { this.setName(name) }
    @computed get age(): number {return this.meta.age}
}

class DogActions {
    @observable activity = {
        walking: false,
        sitting: true
    }
    @action walk() {
        this.activity.sitting = false;
        this.activity.walking = true;
    }
    @action sit() {
        this.activity.walking = false;
        this.activity.sitting = true;
    }
}

class BaseDog {
    alive: boolean = false
    bark() { console.log('WOOF!') }
    kill() { this.alive = false; }
    revive() { this.alive = true; }
}

interface IDog extends BaseDog, DogMeta, DogActions {}

class MyDog extends Mixin<IDog>(BaseDog, DogMeta, DogActions) {
    constructor() {
        super()
    }
    isAwesome(){ console.dir('WOOOOOF!'); }
}
```

Using my regular Mixin code  [`gist`](https://gist.github.com/RobinRadic/085109ca6eee9a471d8b1204bdfec0cd), this doesn't work. MobX sets some properties in each class (like `$mobx` etc).

Currently, im fairly close to having some sort of success, using [`this (gist) mixin`](https://gist.github.com/RobinRadic/8c4431aec26d1d79db158960c7a940d3) code it setis those properties. The problem is, it only sets it properly for the last mixin. So if 2 mixin classes contain `$mobx`, then it's now working as intended.

Now with [`this mixin`](https://gist.github.com/RobinRadic/d2f36df1306e5b86fc125695c8ae4466) i am planning to **somehow** merge those mobx stuff into 1.

- How should i proceed?
- Should i use as different approach?

I want/need to stick with mixins.

Thanks

**Comment by consense** [comment]
@RobinRadic how did you proceed if I may ask?

**Comment by RobinRadic** [comment]
@consense still waiting on this

i haven't used mobx for before (nor react for that matter) but i could probably come up with a way or 2 to realise this, but it would be very, **very** hacky.. 

i'll post it if this issue can't be solved in a fair time

**Comment by gustblima** [comment]
Can you post now? I'm curious haha

**Comment by mweststrate** [comment]
Closing as this is more a generic problem (how to mix in copy class members including there decorators, property accessers etc) then a Mobx problem. Within the current limitations and fluxiness of decorators not something that is going to be addressed by this library soon.
